### PR TITLE
Add a hypothesis fuzz test for consistency with numpy / across backends.

### PR DESCRIPTION
**Pull request by aslvrstn**
Adds a second test that would have caught https://github.com/pytorch/pytorch/issues/86239

The `where` test now passes where it used to fail with
```
    | (shapes (1, 2), (2, 1) mismatch)
    |  x: array([[0, 0]])
    |  y: array([[0],
    |        [0]])
    | Falsifying example: test_where_consistent(
    |     where_inputs=(array([False, False]), array([[0]]), array(0)),
    |     self=<test_consistent_fuzz.TestConsistentWithNumpy testMethod=test_where_consistent>,
    | )
```

before the fix in #86240

The `matmul` test is currently failing locally on MPS, but I'm uncertain whether it is exercising behavior that torch cares to make consistent or not (e.g. numpy produces `inf` while pytorch produces `nan`)

### CODE REVIEW

File: test/test_consistent_fuzz.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+        torch_dev_inputs = [arr.to(device) for arr in torch_inputs]
+        torch_dev_out = torch.matmul(*torch_dev_inputs)
+
+        assert np_out.shape == torch_dev_out.cpu().numpy().shape
+        # Now make sure every backend also produces identical results to numpy (and thus CPU pytorch)
+        # assert_array_almost_equal_nulp(np.nan_to_num(np_out), np.nan_to_num(torch_dev_out.cpu().numpy()), nulp=10000)
+
+    @given(where_inputs())
+    def test_where_consistent(self, device, where_inputs):
```
Comment (mruberry) [inline test/test_consistent_fuzz.py:63]: We already have a test for NumPy consistency:

https://github.com/pytorch/pytorch/blob/07bd053a7ef92263db8d612f4fc7c28e06ade45c/test/test_ops.py#L163

It only runs on the CPU and CUDA device types today, but we could change that

**Comment by mruberry** [comment]
I like the idea here but maybe we should look at extending the existing tests in test files like `test_ops.py` instead of creating new ones?

**Comment by aslvrstn** [comment]
Yea, thank you for that pointer! I was kind of flailing around trying to find the right place for this, and then figured that opening a kinda bad opening gambit PR was probably the best way to get some traction.

I knew there _had_ to be this kind of consistency test somewhere, and I assume it's already doing a better job on CUDA than this matmul test would.

Let me take a stab at turning enough of that test on for MPS to see if it's sufficient to catch the original bug if I regress it manually.

**Comment by aslvrstn** [comment]
Closing in favor of https://github.com/pytorch/pytorch/pull/87342

**Pull request closed without merging**